core/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -153,6 +153,8 @@ public MapperService(IndexSettings indexSettings, IndexAnalyzers indexAnalyzers,
         this.searchQuoteAnalyzer = new MapperAnalyzerWrapper(indexAnalyzers.getDefaultSearchQuoteAnalyzer(), p -> p.searchQuoteAnalyzer());
         this.mapperRegistry = mapperRegistry;
 
+        META_FIELDS = ObjectHashSet.from(mapperRegistry.getMetadataMapperParsers().keySet().toArray(new String[mapperRegistry.getMetadataMapperParsers().keySet().size()]));
```

Comment:
I don't think there is any need to keep `META_FIELDS`, and this is incorrect do both because `META_FIELDS` is static right now (so should not be initialized in the ctor) and because it leaves the incorrect values there now.  `META_FIELDS` should be removed completely, and the methods that use it should instead pull the keyset from `mapperRegistry`.